code/obj/item/mob_parts/robot_parts.dm

```diff
@@ -1067,25 +1068,32 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
 			return
 
 	update_icon()
-		src.overlays = null
-		if(src.chest) src.overlays += image('icons/mob/robots.dmi', "body-" + src.chest.appearanceString, OBJ_LAYER, 2)
-		if(src.head) src.overlays += image('icons/mob/robots.dmi', "head-" + src.head.appearanceString, OBJ_LAYER, 2)
+
+		if(src.chest) src.UpdateOverlays(image('icons/mob/robots.dmi', "body-" + src.chest.appearanceString, FLOAT_LAYER, 2),"chest")
+		else src.UpdateOverlays(null,"chest")
```

Comment:
while you're here, please indent these so it's not a bunch of one liners - it's a bit hard to follow with the multiple nest layers